src/updateDisplay.js

```diff
@@ -0,0 +1,40 @@
+import pressedAnotherOperator from "./pressedAnotherOperator";
+
+const updateDisplay = (result, button, lastEquation) => {
+  const lastCharIsOperator = "/*+-".includes(result[result.length - 1]);
+  const isOperator = "/*+-".includes(button);
+
+  const hasLastChar = (char) => {
+    return result.substr(-1) === char;
+  };
+
+  if (button === "-" && hasLastChar("-")) {
+    button = ""; //no double negatives allowed
+  }
+
+  if (lastEquation !== "" && button !== "=") {
+    //start new equation
+    result = "";
+  }
+
+  if (result === "") {
+    if (isOperator) {
+      result = "0" + button;
+    }
+  }
+
+  if (lastCharIsOperator) {
+    if (".".includes(button)) {
+      result = result + "0.";
+    }
+
+    if (pressedAnotherOperator) {
```

Comment:
@grimxreaper what do you think is happening here?

Reply:
we are fixing this "+." --> "+ 0." 

taking whatever is in result and then adding that bit, if the last character was an operator and the current button is a decimal

Reply:
I mean this line
```
if (pressedAnotherOperator) {
```

Reply:
if `pressedAnotherOperator` returned true...

Reply:
What is the data type of pressedAnotherOperator?

Reply:
Oooohhh! A function!

Reply:
*A boolean

Reply:
Haha. Which is it?

Reply:
Sorry I think I confused both of us. `pressedAnotherOperator` is a function (or object), but it returns a boolean data type.

Reply:
Okay great!